src/Tools/Source/CompilerGeneratorTools/Source/CSharpErrorFactsGenerator/ErrorGenerator.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// We only build the Source Generator in the netstandard target
+#if NETSTANDARD
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Internal.CSharpErrorFactsGenerator
+{
+    [Generator]
+    public sealed partial class ErrorGenerator : IIncrementalGenerator
+    {
+        private readonly struct Model : IEquatable<Model>
+        {
+            public ImmutableArray<string> ErrorNames { get; }
```

Comment:
field?

Reply:
since the wrapper is just a loose wrapper arount an immutable-array, it seems unnecessary.  couldn't the model legit just be the ImmutableArray of strings.  a .Combine on the valuesprovider would work for that, right?

Reply:
sorry, i meant .Collect, not .Combine.